Cron/Write.php

```diff
@@ -2,36 +2,74 @@
 
 namespace WeProvide\NginxRedirect\Cron;
 
-use WeProvide\NginxRedirect\Model\ResourceModel\Redirect\CollectionFactory;
+use Psr\Log\LoggerInterface;
+use WeProvide\NginxRedirect\Model\ResourceModel\Redirect\CollectionFactory as RedirectCollectionFactory;
 use Magento\Framework\App\Config\ScopeConfigInterface;
 
 class Write
 {
+    const MODULE = 'WeProvide_NginxRedirects';
     const PATH = 'weprovide_nginxredirect/path/path';
 
-    protected $collection;
+    protected $redirectCollectionFactory;
     protected $scopeConfig;
+    protected $logger;
 
+    /**
+     * Write constructor.
+     * @param RedirectCollectionFactory $redirectCollectionFactory
+     * @param ScopeConfigInterface $scopeConfig
+     * @param LoggerInterface $logger
+     */
     public function __construct(
-        CollectionFactory $collection,
-        ScopeConfigInterface $scopeConfig
+        RedirectCollectionFactory $redirectCollectionFactory,
+        ScopeConfigInterface $scopeConfig,
+        LoggerInterface $logger
     ) {
-        $this->collection = $collection;
+        $this->redirectCollectionFactory = $redirectCollectionFactory;
         $this->scopeConfig = $scopeConfig;
+        $this->logger = $logger;
     }
 
     public function execute()
     {
-        $redirects = $this->collection->create();
-        $file = $this->scopeConfig->getValue(self::PATH);
+        $redirects = $this->redirectCollectionFactory->create();
 
-        if ($data = fopen($file, "wb")) {
+        if($redirects->count() > 0) {
```

Comment:
What happens if the last redirect is removed? It won't write to the file so the last redirect(s) stay in the list of nginx redirects.

^ note to self